### ISSUE TITLE
Improve Keras Policy Memory Usage

### ISSUE THREAD

**Issue by wochinge** [RasaHQ/rasa#2808]
Training the demo bot with 15 GiB of Ram failed when the training data was shuffled with a `MemoryError`. 

Concrete function `rasa_core/training/data.py`:
```python
    def shuffled_X_y(self):
        idx = np.arange(self.num_examples())
        np.random.shuffle(idx)
        shuffled_X = self.X[idx]
        shuffled_y = self.y[idx]
        return shuffled_X, shuffled_y
```

Ideas:
* use sklearn.utils.shuffle
* do the shuffling as part of the tensorflow graph

**Comment by wochinge** [comment]
`KerasPolicy` was removed in 2.0
